xskillscore/core/probabilistic.py

```diff
@@ -475,6 +478,64 @@ def threshold_brier_score(
         return res.mean(dim, keep_attrs=keep_attrs)
 
 
+def _check_bin_dim(ds):
+    """Assert that bin_dim is in ds. Try to guess and rename edges dimension."""
+    bin_dim = "category_edge"
+    if bin_dim not in ds.dims:
+        raise ValueError(f"require {bin_dim} dimension, found {ds.dims}")
+    return ds
+
+
+def _check_data_within_edges(forecasts, forecasts_edges):
+    """Check that forecasts_edges has category_edge dimension and forecasts_edges span
+    range of forecasts (per variable if xr.Dataset), otherwise ValueError."""
+    if "category_edge" not in forecasts_edges.dims:
+        raise ValueError(
+            "Expect to find category_edge in forecast_edges, found "
+            f"{forecasts_edges.dims}"
+        )
+
+    def _helper_check_data_within_edges(forecasts, forecasts_edges):
+        if forecasts.min() < forecasts_edges.min():
```

Comment:
Are these checks needed if we add the comments to the docstring as below?

I think these will trigger dask computation, so best avoided if possible.

Reply:
its a matter of how much we trust the user on reading the docstring. this function is a security that the inputs are valid. not specifically needed though